docs/features/string-literals-data-section.md

```diff
@@ -0,0 +1,257 @@
+
+[CompilerGenerated]
+internal static class <PrivateImplementationDetails>
+{
+    internal static readonly __StaticArrayInitTypeSize=6 2BBCE396D68A9BBD2517F6B7064666C772694CD92010887BA379E10E7CDDA960 = /* IL: data(48 65 6C 6C 6F 2E) */;
+
+    [StructLayout(LayoutKind.Explicit, Pack = 1, Size = 6)]
+    internal struct __StaticArrayInitTypeSize=6
+    {
+    }
+    
+    internal static class <S>F6D59F18520336E6E57244B792249482
+    {
+        internal static readonly string s;
+
+        unsafe static <S>F6D59F18520336E6E57244B792249482()
+        {
+            s = <PrivateImplementationDetails>.BytesToString((byte*)Unsafe.AsPointer(ref <PrivateImplementationDetails>.2BBCE396D68A9BBD2517F6B7064666C772694CD92010887BA379E10E7CDDA960), 6);
+        }
+    }
+
+    private unsafe static string BytesToString(byte* bytes, int length)
+    {
+        return Encoding.UTF8.GetString(bytes, length);
+    }
+}
+```
+
+### ILVerify and PEVerify
+
+The actual generated code does not call `(byte*)Unsafe.AsPointer`.
+It simply emits `ldsflda` of the data field.
+The same approach is taken by other scenarios like [u8 string literals][u8-literals] and [constant array initializers][constant-array-init].
+That works but does not pass IL/PE verification.
+
+### Ref assemblies
+
+The synthesized types are not part of ref assemblies. That makes them smaller
+and incremental compilation is faster because it does not need to recompile dependent projects when only string literal contents change.
+
+This is automatically implemented, because during metadata-only compilation, method bodies are not emitted,
+and the synthesized types used by this feature are only emitted lazily as part of generating method body instructions.
+
+## Diagnostics
+
+The error CS8103 (Combined length of user strings used by the program exceeds allowed limit) is updated to suggest the feature flag,
+albeit with a disclaimer during the experimental phase of the feature.
+
+## Runtime
+
+Throughput of `ldstr` vs `ldsfld` is very similar (both result in one or two move instructions).
+
+In the `ldsfld` emit strategy, the `string` instances won't ever be collected by the GC once the generated class is initialized.
+`ldstr` has similar behavior, but there are some optimizations in the runtime around `ldstr`,
+e.g., they are loaded into a different frozen heap so machine codegen can be more efficient (no need to worry about pointer moves).
+
+Generating new types by the compiler means more type loads and hence runtime impact,
+e.g., startup performance and the overhead of keeping track of these types.
+
+The generated types are returned from reflection like `Assembly.GetTypes()`
+which might impact the performance of Dependency Injection and similar systems.
+
+> [!NOTE]
+> In practice, there might not be many generated types, it depends on the kind of the program
+> (whether it has lots of short strings or a few large strings) and how [the threshold](#configuration) is configured.
```

Comment:
It looks like the release build of Microsoft.CodeAnalysis.CSharp.dll has roughly 200 strings for file paths, presumably from calls to methods such as `ExceptionUtilities.Unreachable()` that use `[CallerFilePath]`.

Given that, won't many assemblies contain a fair number of strings at 100 characters from paths alone? If so, let's consider removing this `NOTE`.